### PR TITLE
Remove gecko-media dependency

### PR DESCRIPTION
**Pull request by ferjm**
The effort to import Gecko's media stack into Servo has been canceled, so I am removing the bits of code from gecko-media that we added to Servo.

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19806)

### CODE REVIEW

**Comment by tigercosmos** [comment]
I thought dependencies  in `README` also need change?

**Comment by LifeIsStrange** [comment]
By curiosity, could you explain why ?
Importing the media stack would have skyrocketted servo in it's goal of being usable on a daily basis.

**Comment by metajack** [comment]
There were some internal projects that were driving some of the investment there, and those shifted focus a bit. We'd still love to have a media stack in Servo, but the accelerated time frame is no longer viable. In order to keep pushing for media in Servo, it may make sense to use gstreamer as the basis instead, which is already built to be used by many projects.

**Comment by LifeIsStrange** [comment]
@metajack "the investment" , english is not my native language, but if by investment you mean monetary investment, I suggest you , the servo team, to do a kickstarter, your project is exciting by so many ways, not just by the technical challenges (that would motivate many geeks/devs to give) you tackle, but by 1) allowing unprecedent performance on slow devices e.g third world and smartphones/WOT and that would interest many heavy web users.... and by 2) a side effect, allowing new kind of rich websites that would have been too slow before. And that would interest a lot of firms, enabling new markets.

So I don't really think investment would be anymore an issue for servo.

**Comment by jdm** [comment]
By investment, we mean the time spent by Mozilla employees on that project.

**Comment by LifeIsStrange** [comment]
@jdm Yes but new funds from Crowdfunfing could allow to hire new devs maybe ?
Also mozilla could ask to some big firms like Facebook/amazon/Netflix/etc to invest in servo because it's in their interest to have far faster sites. And it has already been done by Facebook for mozjpeg ! They gave 60000 $ to mozilla and mozjpeg has far less economical impact than servo !
Source : http://www.clubic.com/internet/actualite-715615-mozilla-mozjpeg-2-jpeg-plus-leger-soutien-facebook.html (sorry in french).

File: README.md [hunk 1]
```diff
@@ -110,42 +110,27 @@ sudo yum install curl freeglut-devel libtool gcc-c++ libXi-devel \
     freetype-devel mesa-libGL-devel mesa-libEGL-devel glib2-devel libX11-devel libXrandr-devel gperf \
     fontconfig-devel cabextract ttmkfdir python python-virtualenv python-pip expat-devel \
     rpm-build openssl-devel cmake3 bzip2-devel libXcursor-devel libXmu-devel mesa-libOSMesa-devel \
-    dbus-devel ncurses-devel python34 pulseaudio-libs-devel clang clang-libs llvm-toolset-7
+    dbus-devel ncurses-devel python34 pulseaudio-libs-devel
```
Comment (tigercosmos) [inline README.md]: `pulseaudio` also need to remove

File: README.md [hunk 2]
```diff
@@ -85,7 +85,7 @@ sudo apt install git curl freeglut3-dev autoconf libx11-dev \
     libglu1-mesa-dev libgles2-mesa-dev libegl1-mesa-dev \
     pulseaudio dbus-x11 libavcodec-dev libavformat-dev \
     libavutil-dev libswresample-dev  libswscale-dev libdbus-1-dev \
```
Comment (tigercosmos) [inline README.md]: https://github.com/servo/servo/pull/19356/files
Please revert according to this PR

**Comment by ferjm** [comment]
Ugh, I removed too much from the buildbot steps script.

@bors-servo try

**Comment by SimonSapin** [comment]
```
error: failed to load source for a dependency on `webrender_api`

Caused by:
  Unable to update https://github.com/servo/webrender#2547f6fe

Caused by:
  Couldn't rmdir C:\Users\Administrator\.cargo\git\checkouts\webrender-c3596abe1cf4f320\2547f6f

Caused by:
  Access is denied. (os error 5)
```
Are builders stuck in a bad state?

**Comment by SimonSapin** [comment]
Should be fixed now (thanks jdm)

@bors-servo retry

**Pull request merged**